File: Backend/src/main/java/com/example/backend/entity/ShoppingListItemEntity.java
package com.example.backend.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

@Entity
@Table(name = "shopping_list_items")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class ShoppingListItemEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "shopping_list_id", nullable = false)
    private Integer shoppingListId;

    @Column(name = "description", nullable = false, length = 200)
    private String description;

    @Column(name = "is_checked")
    private Boolean isChecked = false;

    @Column(name = "quantity")
    private Double quantity;

    @Column(name = "unit", length = 20)
    private String unit;

}

File: MyApplication/data/src/main/java/com/example/data/Recipes/model/ShoppingListItemEntity.kt
package com.example.data.Recipes.model

import com.example.domain.Recipes.model.ShoppingListItem

data class ShoppingListItemEntity(
    val id: Int,
    val shoppingListId: Int,
    val description: String,
    val isChecked: Boolean = false,
    val quantity: Double? = null,
    val unit: String? = null
) {
    fun toDomain(): ShoppingListItem {
        return ShoppingListItem(
            id = id,
            description = description,
            isChecked = isChecked
        )
    }
}